### ISSUE TITLE
CVE-2022-45688 wrongly still reported for pkg:maven/org.json/json@20230227 due to a bug

### ISSUE THREAD

**Issue by aikebah** [dependency-check/DependencyCheck#5545]
**Describe the bug**
As spotted on

https://github.com/stleary/JSON-java/pull/720#issuecomment-1448044867

CVE-2022-45688 is still reported for `org.json:json` version 20230227

A quick'n'dirty evalution with debug logging indicates that it is because the library version is not properly evaluated, leading to comparing the vulnerable software ranges to a version '*' of the library

```
Check vs cpe:2.3:a:json-java_project:json-java:*:*:*:*:*:*:*:* versions up to (excluding) 20220924 against targetVersion *
```

Note: the NVD entry is currently incorrect regarding the version range, that should've been `up to (excluding) 20230227`, but nevertheless the CVE should not have surfaced on reports for version 20230227

**Version of dependency-check used**
The problem occurs using version 8.1.1 as well as 8.1.2 of the maven plugin

**Log file**
When reporting errors, 99% of the time log file output is required. Please post the log file as a [gist](https://gist.github.com/) and provide a link in the new issue.

**To Reproduce**
1. Create a maven-project with a dependency on the fixed org.json package
    ```xml
    <dependency>
    <groupId>org.json</groupId>
    <artifactId>json</artifactId>
    <version>20230227</version>
    </dependency>
    ```
2. Run `mvn org.owasp:dependency-check-maven:8.1.2:check` on this project
3. Observe that the CVE is still reported

**Expected behavior**
CVE not reported, as the version of the dependency is beyond the vulnerable version range

**Comment by hgueltig** [comment]
As this bug creates a false positive, why not add it to the hosted suppression file until it is fixed to prevent any confusion?

**Comment by ahannah9** [comment]
It seems (`@` into `/`) fixes it for me as it was originally. Maybe this needs to be revisited

Deliberate change commit (`/` into `@`):

https://github.com/jeremylong/DependencyCheck/pull/5504/commits/7d881ee73f2ed7362370029ef1a2ad1bf9da2420

---
main:

https://github.com/jeremylong/DependencyCheck/blob/main/core/src/main/resources/dependencycheck-base-suppression.xml#L6188

https://github.com/jeremylong/DependencyCheck/blob/c0b2c0b505c7d0827c7008cb3602a67d529258f2/core/src/main/resources/dependencycheck-base-suppression.xml#L6188

---
original commit on branch outside repo:

https://github.com/jeremylong/DependencyCheck/blob/8a8e5d202bcfb366f56bd83dddd9bf4ba0e284cb/core/src/main/resources/dependencycheck-base-suppression.xml#L6188

**Comment by aikebah** [comment]
> It seems (`@` into `/`) fixes it for me as it was originally. Maybe this needs to be revisited

No that does not have to be revisited.... it 'fixes' it for you because that original commit was wrong and the current suppression is valid.

The suppression rules is meant to suppress it for anything that is NOT org.json/json

By restoring the original erroneous regEx you are even suppressing it for ANY version of org.json/json (in addition to anything that is not org.json/json), as the packageURL of org.json/json has an @-sign after json, not a forward slash.

**Comment by FyiurAmron** [comment]
note that in the meantime https://nvd.nist.gov/vuln/detail/CVE-2022-45688#VulnChangeHistorySection fixed (changed) the `up to (excluding) 20220924` to `up to (excluding) 20230227` - but, as expected, this false positive still appears.

**Comment by ochstobi** [comment]
Same false positive shows up also in version 20230618. Will this be fixed anytime in the future?

**Comment by aikebah** [comment]
Intention is definitely to get a structural fix for this in the future. I've taking a few looks at it, but it appears to not be an easy fix.

**Comment by chadlwilson** [comment]
Hi @aikebah - think your tactical fix was in `8.4.1` for this one, which resolves this and #5991 - we can probably close both of these now?

I guess the structural fix could be a separate issue?

**Comment by aikebah** [comment]
Right, somehow missed flagging these as resolved
